tests/core/interceptor.c

```diff
@@ -817,3 +818,37 @@ replacement_malloc (gsize size)
 
   return GSIZE_TO_POINTER (size);
 }
+
+TESTCASE (replace_keep_original)
```

Comment:
Would be good to move this so it follows right after the `replace_then_attach`-test. (The test list at the top defines the order that the tests run in. This may differ from the order they're defined in below it. When this is the case, it is because we want lower level tests to run first, as those might detect defects that are much harder to debug in the higher level tests that follow. However, the order of the definitions, i.e. the code, should follow newspaper ordering — the reader should see the higher level view and as they read further down, things get increasingly detailed. This is why e.g. `main()` is always first in a typical Frida program.)